--- a/tw.color.c
+++ b/tw.color.c
@@ -385,16 +385,13 @@
 		    for (i = 0; i < nvariables; i++)
 			if ((Char)variables[i].variable[0] == (v[0] & CHAR) &&
 			    (Char)variables[i].variable[1] == (v[1] & CHAR))
 			    break;
-		    if (i < nvariables) {
 			v += 3;
+			if (i < nvariables)
 			getstring(&c, &v, &variables[i].color, ':');
 			continue;
-		    }
-		    else
-			stderror(ERR_BADCOLORVAR, v[0], v[1]);
 		}
 		break;
 	    }
 	    while (*v && (*v & CHAR) != ':')
 		v++;
