--- a/config.def.h
+++ b/config.def.h
@@ -12,3 +12,4 @@ static const unsigned int borderpx  = 0;        /* border pixel of windows */
 static const unsigned int snap      = 32;       /* snap pixel */
 static const Bool showbar           = 1;     /* False means no bar */
 static const Bool topbar            = 1;     /* False means bottom bar */
+static const unsigned int bottommargin = 12;    /* Margin at the bottom for another bar */

--- a/dwm.c
+++ b/dwm.c
@@ -1841,8 +1841,11 @@ updatebarpos(Monitor *m)
 {
 	m->wy = m->my;
 	m->wh = m->mh;
 	if (m->showbar) {
 		m->wh -= bh;
+		m->wh -= bottommargin;
+		if (!m->topbar)
+			m->wy += bottommargin;
 		m->by = m->topbar ? m->wy : m->wy + m->wh;
 		m->wy = m->topbar ? m->wy + bh : m->wy;
 	} else
