--- dwm.c.orig	2013-07-01 15:12:02.189722419 -0400
+++ dwm.c	2013-07-01 15:21:15.669712092 -0400
@@ -243,6 +243,7 @@ static int textnw(const char *text, unsi
 static void tile(Monitor *);
 static void togglebar(const Arg *arg);
 static void togglefloating(const Arg *arg);
+static void togglescratch(const Arg *arg);
 static void toggletag(const Arg *arg);
 static void toggleview(const Arg *arg);
 static void unfocus(Client *c, Bool setfocus);
@@ -301,6 +302,9 @@ static int gap;
 /* configuration, allows nested code to access above variables */
 #include "config.h"
 
+/* scratchpad gets its own tag :) */
+static unsigned int scratchtag = 1 << LENGTH(tags);
+
 /* compile-time check if all tags fit into an unsigned int bit array. */
 struct NumTags { char limitexceeded[LENGTH(tags) > 31 ? -1 : 1]; };
 
@@ -1170,6 +1174,16 @@ manage(Window w, XWindowAttributes *wa)
 	           && (c->x + (c->w / 2) < c->mon->wx + c->mon->ww)) ? bh : c->mon->my);
 	c->bw = borderpx;
 
+	/* scratchpad */
+	if(strcmp(c->name, scratchpadname) == 0) {
+		c->mon->tagset[c->mon->seltags] |= c->tags = scratchtag;
+		c->isfloating = True;
+		c->x = (c->mon->mw - WIDTH(c)) / 2;
+		c->y = (c->mon->mh - HEIGHT(c)) / 2;
+	}
+	else /* everything else must stay out of scratchtag */
+		c->tags &= TAGMASK;
+
 	wc.border_width = c->bw;
 	XConfigureWindow(dpy, w, CWBorderWidth, &wc);
 	XSetWindowBorder(dpy, w, dc.colors[0][ColBorder].pixel);
@@ -1818,6 +1832,24 @@ togglefloating(const Arg *arg) {
 }
 
 void
+togglescratch(const Arg *arg) {
+	Client *c = NULL;
+	unsigned int found = 0;
+	/* check if a scratchpad is already in the scratchtag */
+	for(c = selmon->clients; c && !(found = c->tags & scratchtag); c = c->next);
+	if(!found) { /* not found: launch it and put it in its tag (see manage()) */
+		spawn(arg);
+		return;
+	}
+	unsigned int newtagset = selmon->tagset[selmon->seltags] ^ scratchtag;
+	if(newtagset) {
+		selmon->tagset[selmon->seltags] = newtagset;
+		arrange(selmon);
+	}
+	focus(c);
+}
+
+void
 toggletag(const Arg *arg) {
 	unsigned int i, newtags;
 
