diff -rc3 a/dwm.c b/dwm.c
*** a/dwm.c	2013-08-15 14:18:25.000000000 +0400
--- b/dwm.c	2013-08-15 14:32:37.450000000 +0400
***************
*** 976,989 ****
  drawtext(const char *text, XftColor col[ColLast], Bool pad) {
  	char buf[256];
  	int i, x, y, h, len, olen;
  
! 	XSetForeground(dpy, dc.gc, col[ColBG].pixel);
! 	XFillRectangle(dpy, dc.drawable, dc.gc, dc.x, dc.y, dc.w, dc.h);
  	if(!text)
  		return;
  	olen = strlen(text);
  	h = pad ? (dc.font.ascent + dc.font.descent) : 0;
! 	y = dc.y + ((dc.h + dc.font.ascent - dc.font.descent) / 2);
  	x = dc.x + (h / 2);
  	/* shorten text if necessary */
  	for(len = MIN(olen, sizeof buf); len && textnw(text, len) > dc.w - h; len--);
--- 976,990 ----
  drawtext(const char *text, XftColor col[ColLast], Bool pad) {
  	char buf[256];
  	int i, x, y, h, len, olen;
+   XftDraw *d;
  
!   XSetForeground(dpy, dc.gc, col[ColBG].pixel);
!   XFillRectangle(dpy, dc.drawable, dc.gc, dc.x, dc.y, dc.w, dc.h);
  	if(!text)
  		return;
  	olen = strlen(text);
  	h = pad ? (dc.font.ascent + dc.font.descent) : 0;
! 	y = dc.y + 1;
  	x = dc.x + (h / 2);
  	/* shorten text if necessary */
  	for(len = MIN(olen, sizeof buf); len && textnw(text, len) > dc.w - h; len--);
***************
*** 992,999 ****
  	memcpy(buf, text, len);
  	if(len < olen)
  		for(i = len; i && i > len - 3; buf[--i] = '.');
  	pango_layout_set_text(dc.plo, text, len);
! 	pango_xft_render_layout(dc.xftdrawable, col + ColFG, dc.plo, x * PANGO_SCALE, y * PANGO_SCALE);
  }
  
  void
--- 993,1002 ----
  	memcpy(buf, text, len);
  	if(len < olen)
  		for(i = len; i && i > len - 3; buf[--i] = '.');
+   d = XftDrawCreate(dpy, dc.drawable, DefaultVisual(dpy, screen), DefaultColormap(dpy,screen));
  	pango_layout_set_text(dc.plo, text, len);
! 	pango_xft_render_layout(d, (XftColor *) &col[ColFG].pixel, dc.plo, x * PANGO_SCALE, y * PANGO_SCALE);
!   XftDrawDestroy(d);
  }
  
  void
***************
*** 1854,1860 ****
  	initfont(font);
  	sw = DisplayWidth(dpy, screen);
  	sh = DisplayHeight(dpy, screen);
! 	bh = dc.h = dc.font.height;
  	updategeom();
  	/* init atoms */
  	wmatom[WMProtocols] = XInternAtom(dpy, "WM_PROTOCOLS", False);
--- 1857,1863 ----
  	initfont(font);
  	sw = DisplayWidth(dpy, screen);
  	sh = DisplayHeight(dpy, screen);
! 	bh = dc.h = dc.font.height + 2;
  	updategeom();
  	/* init atoms */
  	wmatom[WMProtocols] = XInternAtom(dpy, "WM_PROTOCOLS", False);
***************
*** 1994,2000 ****
    buf[ibuf] = 0;
  
  	PangoRectangle r;
! 	pango_layout_set_text(dc.plo, buf, len);
  	pango_layout_get_extents(dc.plo, &r, 0);
  	return r.width / PANGO_SCALE;
  }
--- 1997,2003 ----
    buf[ibuf] = 0;
  
  	PangoRectangle r;
! 	pango_layout_set_text(dc.plo, buf, lenbuf);
  	pango_layout_get_extents(dc.plo, &r, 0);
  	return r.width / PANGO_SCALE;
  }
