diff -rc3 a/dwm.c b/dwm.c
*** a/dwm.c	2013-08-15 14:32:37.000000000 +0400
--- b/dwm.c	2013-08-15 16:13:48.380000000 +0400
***************
*** 315,320 ****
--- 315,325 ----
  static int xerrordummy(Display *dpy, XErrorEvent *ee);
  static int xerrorstart(Display *dpy, XErrorEvent *ee);
  static void zoom(const Arg *arg);
+ static void cycle(const Arg *, int, Bool move);
+ static void cyclenext(const Arg *);
+ static void cycleprev(const Arg *);
+ static void cyclemovenext(const Arg *);
+ static void cyclemoveprev(const Arg *);
  
  /* variables */
  static Systray *systray = NULL;
***************
*** 2635,2640 ****
--- 2640,2698 ----
  	pop(c);
  }
  
+ static void
+ cycle(const Arg *arg, int distance, Bool move)
+ {
+   int i, curtags;
+   int seltag = 0;
+   Arg a;
+ 
+   curtags = selmon->tagset[selmon->seltags] ^ (arg->ui & TAGMASK);
+  for(i = 0; i < LENGTH(tags); i++) {
+    if((curtags & (1 << i)) != 0) {
+      seltag = i;
+      break;
+    }
+   }
+ 
+   seltag = (seltag + distance) % (int)LENGTH(tags);
+   if(seltag < 0)
+     seltag += LENGTH(tags);
+ 
+   a.i = (1 << seltag);
+   if(move){
+     tag(&a);
+     view(&a);
+   }
+   else {
+     view(&a);
+   }
+ }
+ 
+ static void
+ cyclenext(const Arg *arg)
+ {
+   cycle(arg, +1, False);
+ }
+ 
+ static void
+ cycleprev(const Arg *arg)
+ {
+   cycle(arg, -1, False);
+ }
+ 
+ static void
+ cyclemovenext(const Arg *arg)
+ {
+   cycle(arg, +1, True);
+ }
+ 
+ static void
+ cyclemoveprev(const Arg *arg)
+ {
+   cycle(arg, -1, True);
+ }
+ 
  int
  main(int argc, char *argv[]) {
  	if(argc == 2 && !strcmp("-v", argv[1]))
